src/deepsparse/loggers/build_logger.py

```diff
@@ -325,6 +338,52 @@ def possibly_modify_target_identifiers(
     return data_logging_config
 
 
+def metric_functions_from_predefined(
+    metric_functions: List[MetricFunctionConfig], identifier_prefix: str
+) -> Generator[Tuple[str, List[MetricFunctionConfig]], None, None]:
```

Comment:
confused why we would want a generator here, especially since we skip over the first n-1 results. why not just return an overall dict of identifier name mapped to all of the metric functions for it across the built ins specified